README.md

```diff
@@ -2118,6 +2189,58 @@ overrides:
           state: on
           state_not: off
     overrides:
+      cameras:
+        # As this is an array, we need to carefully ensure we are
+        # overridding the correct index. We do this by specifying
+        # earlier indicies as being overridden with an empty object
+        # (in YAML this is `{}`). In this example, overriddes will
+        # only apply to the 2nd camera:
+        - {}                       # No overrides for camera index 0.
```

Comment:
I think this syntax is totally fine. I just also wanted to raise your awareness to ESPHome's array merging strategy, in which they look for items in the array that matches the ID.

```
cameras:
  - id: my_camera
  
overrides:
  cameras:
    - id: my_camera
      <etc>
```

Reply:
Thanks @felipecrs . Could definitely do that also, although it will be strictly harder: Right now, overrides just merge (using https://lodash.com/docs/4.17.15#merge) the two **full** config sets together, which recursively merges all dictionaries and objects, i.e. there is no special treatment for cameras -- the two configs are just smashed together.

Doing something like your (great) ESPHome example, requires us to know in advance which arrays we want to apply the special treatment to, i.e. whilst it might work fine for `cameras`, there are more instances where arrays may need to be merged that don't have an `id` field like this.

It's also something that could be retrofitted later.

Reply:
Makes total sense!